src/guard.rs

```diff
@@ -0,0 +1,90 @@
+use error::{GuardError, ErrorReason};
+use std::mem::align_of;
+
+/// The `Guard` type describes types which define boundary checking strategies.
+pub trait Guard {
```

Comment:
Public API's gonna need examples for all (okay, most) funcitons.

Reply:
Yep, I documented the trait better and added module-level docs with examples.